mica-core/src/main/java/org/obiba/mica/core/upgrade/Mica2Upgrade.java

```diff
@@ -0,0 +1,119 @@
+/*
+ *
+ *  * Copyright (c) 2016 OBiBa. All rights reserved.
+ *  *
+ *  * This program and the accompanying materials
+ *  * are made available under the terms of the GNU Public License v3.0.
+ *  *
+ *  * You should have received a copy of the GNU General Public License
+ *  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.obiba.mica.core.upgrade;
+
+import org.obiba.mica.dataset.HarmonizationDatasetRepository;
+import org.obiba.mica.dataset.StudyDatasetRepository;
+import org.obiba.mica.dataset.domain.HarmonizationDataset;
+import org.obiba.mica.dataset.domain.StudyDataset;
+import org.obiba.mica.dataset.service.HarmonizationDatasetService;
+import org.obiba.mica.dataset.service.StudyDatasetService;
+import org.obiba.mica.network.NetworkRepository;
+import org.obiba.mica.network.domain.Network;
+import org.obiba.mica.network.service.NetworkService;
+import org.obiba.mica.study.StudyRepository;
+import org.obiba.mica.study.domain.DataCollectionEvent;
+import org.obiba.mica.study.domain.Population;
+import org.obiba.mica.study.domain.Study;
+import org.obiba.mica.study.service.StudyService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import java.util.List;
+
+@Component
+public class Mica2Upgrade {
+
+  private static final Logger logger = LoggerFactory.getLogger(Mica2Upgrade.class);
+
+  @Autowired
```

Comment:
It's not a problem but the whole code bases uses @Inject, may be you could use the same thing